src/parser.js

```diff
@@ -818,74 +800,94 @@ export default class Parser {
     }
 
     splitWord (namespace, firstCallback) {
+        // Collect all the relevant tokens together
+        const tokensList = [this.currToken];
         let nextToken = this.nextToken;
-        let word = this.content();
         while (
             nextToken &&
             ~[tokens.dollar, tokens.caret, tokens.equals, tokens.word].indexOf(nextToken[TOKEN.TYPE])
         ) {
             this.position ++;
-            let current = this.content();
-            word += current;
-            if (current.lastIndexOf('\\') === current.length - 1) {
+            const token = this.currToken;
+            tokensList.push(token);
+            if (this.content(token).endsWith('\\')) {
```

Comment:
This should be equivalent to `current.lastIndexOf('\\') === current.length - 1`, apart from the case when `current` is equal to `""`. I think we never get empty strings here though, but let me know if I'm wrong.

Reply:
But also I'm not sure if this condition is ever true. The coverage report says that this if statement never evaluates to true during tests (both in this PR and also on base), and I haven't been able to write a test that hits this.

I tried parsing `.#{$classname1}\\$classname2`, but it gets tokenized as `["/#{", "$", "classname1}\\$classname2"]` and not as `["/#{", "$", "classname1}\\", "$", "classname2"]`, so maybe the tokenizer handles this already?